go-cosmwasm/types/queries.go

```diff
@@ -289,3 +306,31 @@ func (d *RewardCoins) UnmarshalJSON(data []byte) error {
 	*d = raw
 	return nil
 }
+
+// MarshalJSON ensures that we get [] for empty arrays
+func (d ProposalsResponse) MarshalJSON() ([]byte, error) {
+	if len(d.Proposals) == 0 {
+		return []byte("{\"proposals\": []}"), nil
+	}
+	var raw = d.Proposals
+	asBytes, err := json.Marshal(raw)
+	if err != nil {
+		return nil, err
+	}
+
+	return append(append([]byte("{\"proposals\": "), asBytes...), []byte("}")...), nil
+}
+
+// UnmarshalJSON ensures that we get [] for empty arrays
+func (d *ProposalsResponse) UnmarshalJSON(data []byte) error {
+	// make sure we deserialize [] back to null
+	if string(data) == "{\"proposals\": []}" || string(data) == "null" {
+		return nil
+	}
```

Comment:
What do you mean by "deserialize the json"? This is exactly the function that tells the code how to deserialize the json. This due to a difference with how empty arrays are serialized in rust vs in go. I'll throw in a check without the whitespace as well, but this function is never called anyway so....

Also, this is a method on `ProposalsResponse` already so I don't understand the last part...

Reply:
I meant that you're doing string comparison on serialized jsons, and it's more precise to turn the json into the Go equivalent of `serde_json::Value` and do the comparison on that. If you'll have inputs coming in from new places with different whitespaces, it'll break.

As for the second part - this function is a method on instances of `ProposalsResponse` rather than a true constructor. if you have an instance of this type that was already initialized, calling this method on it would overwrite the previous values it held.

Reply:
There's also no equivalent of `serde_json::Value`. You could maybe use `map[string]interface{}`, but that's pretty weird, and honestly, I don't care if this fails because we _never use this function_

Also this is the idiomatic way to deserialize json values in Go afaik

Reply:
Oh well.
tbh you can probably just delete this if it isn't used?
resolving anyways.